### PR TITLE
Create, tested and fixed function for users to unsave a job.

### PR DESCRIPTION
**Pull request by Rdore88**

### CODE REVIEW

File: app/controllers/api/saved_jobs_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Api::SavedJobsController < ApplicationController
-  before_action :authorize!, only: [:index, :show, :create]
+  before_action :authorize!, only: [:index, :show, :create, :destroy]
```
Comment (rposborne) [inline app/controllers/api/saved_jobs_controller.rb]: Are any of the endpoints in this controller unauthenticated.  If they aren't we can drop the only.

File: app/controllers/api/saved_jobs_controller.rb [hunk 2]
```diff
@@ -26,6 +26,15 @@ def show
     end
   end
 
+  def destroy
+    job_connection = SavedJob.joins(:job).find_by("jobs.job_key = '#{params[:key]}'", "user_id = #{current_user.id}")
```
Comment (rposborne) [inline app/controllers/api/saved_jobs_controller.rb]: This seems really complex and includes sql injection issues.  (params direct into a sql statement are a major no no" Instead use ? mark as a placeholder and the arity of the find_by function will increase per ?

File: app/controllers/api/saved_jobs_controller.rb [hunk 3]
```diff
@@ -26,6 +26,15 @@ def show
...
+    if job_connection.destroy
+      render json: {message: "Job is no longer saved", status: :ok}
+    else
+      render json: {message: @job_connection.errors.full_message, status: :not_found}
```
Comment (rposborne) [inline app/controllers/api/saved_jobs_controller.rb]: Has this been tested?  Pretty sure it's `full_messages`

**Review by rposborne — CHANGES_REQUESTED** [review]
Security issue around sql statement.

**Review by rposborne — APPROVED** [review]
🚢

**Comment by jenniferoakes** [comment]
🚢

**Pull request merged**